unit_tests/system/profiler_test.cpp

```diff
@@ -9,69 +9,70 @@
 #include <cstdlib>
 #include <sstream>
 
-
-
 #define TEST_USE_MPI
+#define TEST_USE_PETSC
 #include <flow_gtest_mpi.hh>
 
-
-
+/**
+ * Fixture class for testing Profiler protected members
+ */
+class ProfilerTest: public testing::Test {
+    public:
+        void test_str_hash();
```

Comment:
Main problem with replacing macros is following:
 `ProfilerTest` must be defined before `Profiler` (so it can be marked as friend)
but macros which should be replaced are working with unknown types since `Timer` is not included yet because it is in the same file as `Profiler`. These condition can never be met as long as both classes remain in the same file.

Since we are dealing with macros which only makes test more organized I suggest to keep macros for now but when we detach profiler to its own project I suggest to separate classes so we can have options when including.